src/compiler/checker.ts

```diff
@@ -49029,7 +49030,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function calculateNodeCheckFlagWorker(node: Node, flag: LazyNodeCheckFlags) {
-        if (!compilerOptions.noCheck) {
+        if (!compilerOptions.noCheck || !canIncludeBindAndCheckDiagnsotics(getSourceFileOfNode(node), compilerOptions)) {

```

Comment:
```suggestion
        if (!compilerOptions.noCheck && !canIncludeBindAndCheckDiagnsotics(getSourceFileOfNode(node), compilerOptions)) {

```

This difference will only surface as a performance issue, since we'll calculate flags when we don't need to :)